File: programmers/lv1/덧칠하기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/161989

function solution(n, m, section) {
  let answer = 0;
  let wall = new Array(n).fill(1);
  section.forEach((e) => (wall[e - 1] = 0));
  while (wall.includes(0)) {
    let len = wall.length;
    for (let i = 0; i < len; i++) {
      if (!wall[i]) {
        for (let j = 0; j < m; j++) {
          if (!wall[i + j]) wall[i + j] = 1;
        }
        answer++;
      }
    }
  }
  return answer;
}

File: programmers/lv1/덧칠하기.py
def solution(n, m, section):
    answer = 0
    painted = [1] * n
    for s in section:
        painted[s-1] = 0
        
    while painted.count(0):
        for i in range(len(painted)):
            if not painted[i]:
                painted[i:i+m] = [1] * m
                answer += 1
                
    return answer
